core/src/main/java/org/apache/carbondata/core/scan/executor/util/RestructureUtil.java

```diff
@@ -165,14 +165,15 @@ private static boolean isColumnMatches(boolean isTransactionalTable,
     // column ID but can have same column name
     if (tableColumn.getDataType().isComplexType() && !(tableColumn.getDataType().getId()
         == DataTypes.ARRAY_TYPE_ID)) {
-      if (tableColumn.getColumnId().equals(queryColumn.getColumnId())) {
+      if (tableColumn.getColumnId().equalsIgnoreCase(queryColumn.getColumnId())) {
```

Comment:
I think there are many other place columnID check is there. example filter scenario. 
If handle this way need to check and handle all the places.

I suggest better to convert inferred schema to lower case at the place of inferring. That way no need to handle all these places